x/dsmr/block.go

```diff
@@ -105,6 +106,47 @@ func ParseChunk[T Tx](chunkBytes []byte) (Chunk[T], error) {
 	return c, c.init()
 }
 
+// ExecutionBlock bridge the gap between the dsmr's block implementation and the validity window's execution block interface.
+type ExecutionBlock struct {
+	innerBlock Block
+	certSet    set.Set[ids.ID]
+}
+
+func (e ExecutionBlock) Timestamp() int64 {
+	return e.innerBlock.Timestamp
+}
+
+func (e ExecutionBlock) Height() uint64 {
+	return e.innerBlock.Height
+}
+
+func (e ExecutionBlock) Contains(id ids.ID) bool {
+	return e.certSet.Contains(id)
+}
+
+func (e ExecutionBlock) Parent() ids.ID {
+	return e.innerBlock.ParentID
+}
+
+func (e ExecutionBlock) Containers() []*emapChunkCertificate {
+	emapChunkCert := make([]*emapChunkCertificate, len(e.innerBlock.ChunkCerts))
```

Comment:
nit: rename to `chunkCerts`. Generally prefer that slices/maps/etc have plural names.